### PR TITLE
Add recs pagination in header and footer

### LINKED ISSUE #453: Duplicate pagination on Recs Table to also show above
**Context:**
In the case that a user has a large number of rows in view, it is hard to see what page you are on when pagination is just at the bottom. Especially if you apply a filter, it can be unclear. We would like to improve this experience.

**Out of scope**
- already completed: pagination at bottom of table

**A/C's**
- [ ] When I view the recommendations table, I see a pagination component at the top of the recommendations table, but below the metric tons/kg toggle

### PR DESCRIPTION
**Pull request by jgrumboe**
## Description of Change

This adds pagination in the header and is a retry of #462.
I like it as my first Typescript contribution. 

It will solve #453.

## Checklist

- [x] PR description included and stakeholders cc'd
- [ ] tests are changed or added
- [x] yarn test passes
- [x] yarn lint has been run
- [x] git pre-commit hook is successfully executed.
      (**Please refrain from using `--no-verify`**)
- [ ] yarn changeset was run and relevant packages have been updated as a major/minor/patch bump with descriptions
      (**Determine version update using [Semantic Versioning](https://semver.org/)**)
- [ ] relevant documentation is changed or added

### CODE REVIEW

**Comment by jgrumboe** [comment]
I'm somehow stuck here and there are two things left, where I need help.
1. I think I managed to separate the `CustomPagination` into its own file but now get a TS warning:
  ```
/Users/jgrumboeck/git/jgrumboe/cloud-carbon-footprint/packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx
  16:10  warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

✖ 1 problem (0 errors, 1 warning)
  ```
  Don't know how to get rid of it.

2. I didn't manage to refactor the function parameters into props. Tried but didn't find a good example for me I could understand and therefore don't know what to change. Sorry.

**Comment by jgrumboe** [comment]
@ericksod
@4upz-tw

File: packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  GridToolbarContainer,
+} from '@material-ui/data-grid'
+import useStyles from './recommendationsTableStyles'
+import Pagination from '@material-ui/lab/Pagination'
+import { Box, MenuItem, Select, Typography } from '@material-ui/core'
+
+function CustomPagination(
+  handlePageSizeChange: (event: React.ChangeEvent<{ value: unknown }>) => void,
+) {
+  return () => {
```
Comment (4upz-tw) [inline packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx]: This duplicated return can be removed. You only need one which is on line 20.

File: packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  useGridSlotComponentProps,
+  GridToolbarContainer,
+} from '@material-ui/data-grid'
+import useStyles from './recommendationsTableStyles'
+import Pagination from '@material-ui/lab/Pagination'
+import { Box, MenuItem, Select, Typography } from '@material-ui/core'
+
+function CustomPagination(
+  handlePageSizeChange: (event: React.ChangeEvent<{ value: unknown }>) => void,
+) {
```
Comment (4upz-tw) [inline packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx]: You can add a return type for this component called `ReactElement` which is imported from `React`. 

```suggestion
function CustomPagination(
  handlePageSizeChange: (event: React.ChangeEvent<{ value: unknown }>) => void,
): ReactElement {
```

File: packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx [hunk 3]
```diff
@@ -209,9 +216,11 @@ const RecommendationsTable: FunctionComponent<RecommendationsTableProps> = ({
               }}
               onRowClick={handleRowClick}
               disableColumnFilter
-              pagination
               pageSize={pageState.pageSize}
-              rowsPerPageOptions={[25, 50, 100]}
+              components={{
+                Toolbar: CustomPagination(handlePageSizeChange),
+                Pagination: CustomPagination(handlePageSizeChange),
```
Comment (4upz-tw) [inline packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx]: To fix the double return and the type issue in the Pagination file, the assigned components can be changed to return ReactElements as followed:

```suggestion
                Toolbar: () => CustomPagination(handlePageSizeChange),
                Pagination: () => CustomPagination(handlePageSizeChange),
```

**Review by 4upz-tw — CHANGES_REQUESTED** [review]

**Comment by 4upz-tw** [comment]
@jgrumboe Thanks for continuing to work on this! It looks and works good! Here are some responses to your concerns:

1. If you submit an additional commit with the suggested changes from my review, it should fix that TS warning you were referencing. After you commit those changes, I can merge everything in and close this out 😎

2. No worries at all! We can always refactor it after. If you'd like, I can commit the change after the merge and link you to the commit so you can see an example and hopefully make more sense of what I meant.

File: packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx (hunk 1 above)
Comment (jgrumboe) [reply packages/client/src/pages/RecommendationsPage/RecommendationsTable/CustomPaginaton.tsx]: ok

File: packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx (hunk 3 above)
Comment (jgrumboe) [reply packages/client/src/pages/RecommendationsPage/RecommendationsTable/RecommendationsTable.tsx]: ok

**Comment by jgrumboe** [comment]
@4upz-tw 
Hi Arik,
Thanks for your comments. I changed and pushed it recently.
And yes, please commit the refactor changes and link me to the commit.
Cheers, Johannes

**Comment by jgrumboe** [comment]
btw: regarding Hacktoberfest it would be nice to get a "hacktoberfest-accepted" label 😊

**Review by 4upz-tw — APPROVED** [review]
Looks good! Thanks again, @jgrumboe! I'll go ahead and merge this in, and of course, add the hacktoberfest-accepted label to it 🎉

**Pull request merged**